lib/include/openamp/virtio.h

```diff
@@ -111,6 +111,24 @@ struct virtio_buffer_info {
 	unsigned long size;
 };
 
+/**
+ * struct remoteproc_vring - remoteproc vring structure
+ * @vq virtio queue
+ * @va logical address
+ * @notifyid vring notify id
+ * @num_descs number of descriptors
+ * @align vring alignment
+ * @io metal I/O region of the vring memory, can be NULL
+ */
+struct virtio_vring {
+	struct virtqueue *vq;
+	void *va;
+	uint32_t notifyid;
+	unsigned int num_descs;
```

Comment:
Are num_descs & align really needed?  Seems like these are just init params - so maybe we need a virtio_init() or something

Reply:
it is used for vring initialization through virtqueue initialization. the resource table has this information. i am more prone to have this information and the virtqueue initialization which is the virtio_create_virqueues() function in the remoteproc_virtio in the staging branch.